src/app/AlwaysOn.HealthService/ComponentHealthChecks/AzMonitorHealthScoreCheck.cs

```diff
@@ -0,0 +1,69 @@
+﻿using AlwaysOn.Shared;
+using Azure;
+using Azure.Identity;
+using Azure.Monitor.Query;
+using Azure.Monitor.Query.Models;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace AlwaysOn.HealthService.ComponentHealthChecks
+{
+    public class AzMonitorHealthScoreCheck : IHealthCheck
+    {
+        private readonly ILogger<AzMonitorHealthScoreCheck> _log;
+        private readonly SysConfiguration _sysConfig;
+        private readonly LogsQueryClient _logsQueryClient;
+
+        public AzMonitorHealthScoreCheck(ILogger<AzMonitorHealthScoreCheck> log,
+            SysConfiguration sysConfig)
+        {
+            _log = log;
+            _sysConfig = sysConfig;
+
+            _logsQueryClient = new LogsQueryClient(new DefaultAzureCredential(new DefaultAzureCredentialOptions()
+            {
+                ManagedIdentityClientId = _sysConfig.ManagedIdentityClientId
+            }));
+        }
+
+        /// <summary>
+        /// Query regional Log Analytics workspace to fetch the latest HealthScore
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                Response<LogsQueryResult> response = await _logsQueryClient.QueryWorkspaceAsync(
+                    _sysConfig.RegionalLogAnalyticsWorkspaceId,
+                    _sysConfig.HealthServiceAzMonitorHealthScoreQuery,
+                    new QueryTimeRange(TimeSpan.FromMinutes(10)),
+                    cancellationToken: cancellationToken);
+
+                LogsTable table = response.Value.Table;
+
+                foreach (var row in table.Rows) // there will be only one row (take 1)
```

Comment:
hmm good point. Now that the query is outside in the Config, this makes it even harder to understand. But how else should we do it? currently we just expect that the query must only return one row (or we ignore everything else). any ideas?

Reply:
Is it ignoring other rows? The way I read the code it will iterate through all rows until it finds one which is not healthy (then returns `Unhealthy`), or returns `Healthy` if all rows are above threshold.